docs/content/server/new_blocks.md

```diff
@@ -109,64 +122,46 @@ To create a new block, follow these steps:
 
-            # Provide sample input and output for testing the block
+## Tips for Effective Block Testing
 
-            test_input={"topic": "Artificial Intelligence"},
-            test_output={"summary": "Artificial intelligence (AI) is intelligence demonstrated by machines, in contrast to the natural intelligence displayed by humans and animals."},
-        )
+1. **Provide realistic test_input**: Ensure your test input covers typical use cases.
 
-    def run(self, input_data: Input) -> BlockOutput:
-        try:
-            # Make the request to Wikipedia API
-            response = requests.get(f"https://en.wikipedia.org/api/rest_v1/page/summary/{input_data.topic}")
-            response.raise_for_status()
-            summary_data = response.json()
+2. **Define appropriate test_output**: 
+   - For deterministic outputs, use specific expected values.
+   - For non-deterministic outputs or when only the type matters, use Python types (e.g., `str`, `int`, `dict`).
+   - You can mix specific values and types, e.g., `("key1", str), ("key2", 42)`.
 
-            # Output the summary
-            yield "summary", summary_data['extract']
+3. **Use test_mock for network calls**: This prevents tests from failing due to network issues or API changes.
 
-        except requests.exceptions.HTTPError as http_err:
-            raise ValueError(f"HTTP error occurred: {http_err}")
-        except requests.RequestException as e:
-            raise ValueError(f"Request to Wikipedia API failed: {e}")
-        except KeyError as e:
-            raise ValueError(f"Error processing Wikipedia data: {e}")
-```
+4. **Consider omitting test_mock for blocks without external dependencies**: If your block doesn't make network calls or use external resources, you might not need a mock.
 
-## Key Points to Remember
+5. **Consider edge cases**: Include tests for potential error conditions in your `run` method.
 
-- **Unique ID**: Give your block a unique ID in the `__init__` method.
-- **Input and Output Schemas**: Define clear input and output schemas.
-- **Error Handling**: Implement error handling in the `run` method.
-- **Output Results**: Use `yield` to output results in the `run` method.
-- **Register the Block**: Add your new block to the `__init__.py` file to make it available to the server.
-- **Testing**: Provide test input and output in the `__init__` method for automatic testing.
+6. **Update tests when changing block behavior**: If you modify your block, ensure the tests are updated accordingly.
 
-By following these steps, you can create new blocks that extend the functionality of the AutoGPT Agent Server.
+By following these guidelines, you can create robust, testable blocks that enhance the AutoGPT Agent Server's capabilities while ensuring reliability and consistency, even for blocks with non-deterministic outputs.
```

Comment:
Silly unrelated questions: is this AI generated ? :D

Reply:
haha I had claude go over it with me and i missed this, will put that back to how it was